Vester/Private/Template/VesterTemplate.Tests.ps1

```diff
@@ -22,18 +22,50 @@ Param(
     [switch]$Remediate
 )
 
-# Process .Vester.ps1 files one at a time
-ForEach ($Test in $TestFiles) {
-    Write-Verbose "Processing test file $Test"
-    $TestName = Split-Path $Test -Leaf
-    $Scope = Split-Path (Split-Path $Test -Parent) -Leaf
+# Gets the scope, the objects for the scope and their requested test files
+$Scopes = Split-Path (Split-Path $TestFiles -Parent) -Leaf | Select -Unique
+$Final = @()
+$InventoryList = @()
+$Datacenter = Get-Datacenter -Name $cfg.scope.datacenter -Server $cfg.vcenter.vc
+foreach($Scope in $Scopes)
+{
+    Write-Verbose "Processing $Scope"
+    Remove-Variable InventoryList -ErrorAction SilentlyContinue # Makes sure the variable is always fresh
+    # Use $Scope (parent folder) to get the correct objects to test against
+    # If changing values here, update the "$Scope -notmatch" test below as well
+    $InventoryList = switch ($Scope) {
+        'vCenter'    {$cfg.vcenter.vc}
+        'Datacenter' {$Datacenter}
+        'Cluster'    {$Datacenter | Get-Cluster -Name $cfg.scope.cluster}
+        'DSCluster'  {$Datacenter | Get-DatastoreCluster -Name $cfg.scope.dscluster}
+        'Host'       {$Datacenter | Get-Cluster -Name $cfg.scope.cluster | Get-VMHost -Name $cfg.scope.host}
+        'VM'         {$Datacenter | Get-Cluster -Name $cfg.scope.cluster | Get-VM -Name $cfg.scope.vm}
+        'Network'    {$Datacenter | Get-VDSwitch -Name $cfg.scope.vds}
+    }
+
+    $ScopeObj = [pscustomobject] @{
+        'Scope'         = $Scope
+        'InventoryList' = $InventoryList
+        'TestFiles'     = $TestFiles | Where-Object { (Split-Path (Split-Path $_ -Parent) -Leaf) -eq $Scope }
+    }
+    if (($ScopeObj.InventoryList -ne $NULL) -and ($ScopeObj.TestFiles -ne $NULL)){
+        $Final += $ScopeObj
+    }
+}
+
+# Loops through each Scope
+foreach($Scope in $Final.Scope)
+{
+    # Pulling the inventory and test files for this scope
+    $Inventory = ($Final | Where-Object { $_.Scope -eq $Scope }).InventoryList
+    $Tests = ($Final | Where-Object { $_.Scope -eq $Scope }).TestFiles
 
     # The parent folder must be one of these names, to help with $Object scoping below
     # If adding here, also needs to be added to the switch below
-    If ($Scope -notmatch 'vCenter|Datacenter|Cluster|DSCluster|Host|VM|Network') {
+    If ('vCenter|Datacenter|Cluster|DSCluster|Host|VM|Network' -notmatch $Scope) {
         Write-Warning "Skipping test $TestName. Use -Verbose for more details"
         Write-Verbose 'Test files should be in a folder with one of the following names:'
-        Write-Verbose 'vCenter / Datacenter / Cluster / DSCluster / Host / VM / Network'
+        Write-Verbose 'vCenter / Datacenter / Cluster / Host / VM / Network'
```

Comment:
Thanks for catching that. DSCluster was missing by mistake.
I have added it back in.